src/codecs/timidity/playmidi.c

```diff
@@ -560,6 +560,75 @@ static void seek_forward(MidiSong *song, Sint32 until_time)
   song->current_sample=until_time;
 }
 
+static void seek_forward_beat(MidiSong *song, Uint32 beat)
+{
+  reset_voices(song);
+  while (song->current_event->beat < beat)
+    {
+      switch(song->current_event->type)
```

Comment:
Exact same contents in this switch statement as the other seek_foward function. I could split out to a help function.